main.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+const head = document.getElementById('head');
+const h2 = document.querySelector('h2');
+const h3 = document.querySelector('h3');
+const inner = document.getElementById('inner');
+const ul = document.querySelector('ul');
+const startButton = document.getElementById('start-button');
+let randomData = {};
+let count = 1;
+let correctCounter = 0;
+
+
+const asynchronousFunc = () => {
+  return new Promise((resolve, reject) => {
+    fetch('https://opentdb.com/api.php?amount=50&type=multiple')
+    .then((response) => {
+      return resolve(response.json())
+    }).catch(() => {
+      return reject('失敗')
+    })
+  })
+};
```

Comment:
2点ほど指摘があります。

1. 名前が抽象的すぎる
日本語訳すると「非同期な関数」になりますが、これだと「具体的に何をしてくれる関数」か読み取れないと思います。

2. Promise を使う書き方もよいですが `async/await` を使うと同期処理とコードの書き方が近くなるのでお勧めです。

```suggestion
const getQuizJson = async () => {
  try {
    const response = await fetch('https://opentdb.com/api.php?amount=50&type=multiple')
    return await response.json();
  } catch (e) {
    console.error(e);
  }
};
```